### PR TITLE
Formalize most public API

### PR DESCRIPTION
**Pull request by PepperCode1**
This pull request performs many, mostly minor, changes across most public API to make it nicer to use and more stable in general.

### CODE REVIEW

File: fabric/src/lib/java/dev/engine_room/flywheel/lib/model/baked/BakedModelBufferer.java [hunk 1]
```diff
@@ -31,8 +32,18 @@ final class BakedModelBufferer {
...
+		if (level == null) {
+			if (state == null) {
+				state = Blocks.AIR.defaultBlockState();
+			}
+			FabricOriginBlockAndTintGetter originLevel = objects.level;
+			originLevel.originBlockState(state);
+			level = originLevel;
+		} else if (state == null) {
+			state = level.getBlockState(BlockPos.ZERO);
+		}
```
Comment (Jozufozu) [inline fabric/src/lib/java/dev/engine_room/flywheel/lib/model/baked/BakedModelBufferer.java:46]: This seems like complex behavior to hide in the impl. Somewhat related, perhaps there should be explicit methods for defaultLevel and defaultBlockState in the common builder

File: common/src/lib/java/dev/engine_room/flywheel/lib/model/baked/PartialModel.java [hunk 2]
```diff
@@ -1,53 +1,49 @@
...
+	@UnknownNullability
+	BakedModel bakedModel;
 
+	private PartialModel(ResourceLocation modelLocation) {
 		this.modelLocation = modelLocation;
 
-		synchronized (ALL) {
-			ALL.add(this);
+		if (populateOnInit) {
+			FlwLibXplat.INSTANCE.getBakedModel(Minecraft.getInstance().getModelManager(), modelLocation);
```
Comment (Jozufozu) [inline common/src/lib/java/dev/engine_room/flywheel/lib/model/baked/PartialModel.java]: Looks like bakedModel should be assigned here, and probably throw an error if it's missing

**Review by Jozufozu — CHANGES_REQUESTED** [review]
Lots of good stuff in here, just a few things that caught my eye

File: common/src/lib/java/dev/engine_room/flywheel/lib/model/baked/PartialModel.java (hunk 2 above)
Comment (PepperCode1) [reply common/src/lib/java/dev/engine_room/flywheel/lib/model/baked/PartialModel.java]: Good catch. The method returns the missing model if there were errors during loading or the model doesn't exist. In the former case, vanilla already logs a warning/error, so an additional error isn't necessary.

Comment (Jozufozu) [reply common/src/lib/java/dev/engine_room/flywheel/lib/model/baked/PartialModel.java]: Ah a missing model would be better yeah

File: fabric/src/lib/java/dev/engine_room/flywheel/lib/model/baked/BakedModelBufferer.java (hunk 1 above)
Comment (PepperCode1) [reply fabric/src/lib/java/dev/engine_room/flywheel/lib/model/baked/BakedModelBufferer.java:46]: All builder methods for all three types of builders are optional and thus have default values. Would it be better to add `default*()` methods for all properties or accept nullable parameters, where null means default?

**Review by Jozufozu — APPROVED** [review]
Very nice changes ❤️

**Comment by PepperCode1** [comment]
I will merge the PR now as there are a lot of changes and waiting for more minor changes is not worth the merge conflicts with other mainline changes that this will cause.
Things to review later:
- Complex level and state behavior in model builders
- Default methods for model builders
- Adding or removing data from `RenderContext`; passing a reduced `RenderContext` to `Engine#updateRenderOrigin` with more than just the `Camera`
- More registry refactors, including renames, removing `registerAndGet`, splitting `MaterialShaders`, making shader source locations easier to register, and refactoring and freezing `VisualizerRegistry`

**Pull request merged**